### PR TITLE
Rails 6.1 Compatibility

### PR DESCRIPTION
**Pull request by andrew-newell**
I'll be completely honest, this fix works, but I haven't spent the time to guarantee this is the "right" way to update this library for Rails 6.1 compatibility. Due to the way the tests fiddle with the innards of ActiveRecord, it seems like this library is only a few more major Rails releases away from needing its tests rewritten.

I saw the comments on https://github.com/voormedia/rails-erd/pull/345 by @gee-forr and several others, hopefully I'll be able to get the tests in travis happy if/when they fail.

### CODE REVIEW

**Comment by andrew-newell** [comment]
@kerrizor I think this is ready for review now :)

**Comment by katelovescode** [comment]
Would love to have this merged!  I've got the fork's master branch pointed to in my Gemfile, thanks for this fix @andrew-newell !

File: lib/rails_erd/domain.rb [hunk 1]
```diff
@@ -49,7 +49,13 @@ def initialize(models = [], options = {})
     # Returns the domain model name, which is the name of your Rails
     # application or +nil+ outside of Rails.
     def name
-      defined? Rails and Rails.application and Rails.application.class.parent.name
+      return unless defined?(Rails) && Rails.application
```
Comment (dmuneras) [inline lib/rails_erd/domain.rb:52]: ❤️

File: lib/rails_erd/domain.rb [hunk 2]
```diff
@@ -49,7 +49,13 @@ def initialize(models = [], options = {})
...
+
+      if Rails.application.class.respond_to?(:module_parent)
```
Comment (dmuneras) [inline lib/rails_erd/domain.rb:54]: ❤️

**Comment by homonecloco** [comment]
I also would love to have this merged and released! I just discovered the library, and patched my local installation

**Review by dmuneras — APPROVED** [review]
Wondering why I can approve this PR 😅 , but I will give it a try.

**Comment by andrew-newell** [comment]
> Wondering why I can approve this PR 😅 , but I will give it a try.

I like your efforts, unfortunately I don't think you or I have the permissions to merge lol

**Comment by dmuneras** [comment]
> > Wondering why I can approve this PR 😅 , but I will give it a try.
> 
> I like your efforts, unfortunately I don't think you or I have the permissions to merge lol

Just trying to get some attention from people that can 😂

**Comment by homonecloco** [comment]
> > > Wondering why I can approve this PR 😅 , but I will give it a try.
> > 
> > 
> > I like your efforts, unfortunately I don't think you or I have the permissions to merge lol
> 
> Just trying to get some attention from people that can 😂

Same reason why I commented!

**Comment by tinacious** [comment]
I'm on 6.1.0 and can confirm that this fixes the error. Thank you.

**Comment by 0bserver07** [comment]
Ah thanks for the PR,  a quick hack here until someone responds:

```

mkdir lib/rails_erd/

touch lib/rails_erd/domain.rb

```

copy the commit content from this file [here](https://raw.githubusercontent.com/voormedia/rails-erd/cd225439688b6a9f7d74f28106fe2ada73090345/lib/rails_erd/domain.rb) to below file:

```
ib/rails_erd/domain.rb
```

**Comment by katelovescode** [comment]
🤖 💡 Just wanting to keep this bumped because I don't want to see it get auto-closed or anything!

File: test/unit/attribute_test.rb [hunk 3]
```diff
@@ -266,14 +266,13 @@ def create_attribute(model, name)
...
-      define_method :limit do
-        # https://github.com/voormedia/rails-erd/issues/21
-        { :srid => 4326, :type => "point", :geographic => true }
-      end
+    create_model "Business", :location => :integer do
+    define_singleton_method :limit do
+      # https://github.com/voormedia/rails-erd/issues/21
+      { :srid => 4326, :type => "point", :geographic => true }
     end
+  end
```
Comment (kerrizor) [inline test/unit/attribute_test.rb:274]: I feel like we've got an indentation issue or missing `end` here?

Comment (andrew-newell) [reply test/unit/attribute_test.rb]: Yep, looks like the indentation was wrong, should be fixed now!

Comment (kerrizor) [reply test/unit/attribute_test.rb]: Yep.. I'll fix it in a follow-up MR

**Comment by kerrizor** [comment]
@andrew-newell Looks good! I'm honestly surprised the tests continue to pass for 4.2, but it's probably time to EOL support for that. CI comes back 🍏 I'll merge and push a new point release out.. Thank you for the contribution! 🏆

**Comment by kerrizor** [comment]
v1.6.1 pushed to Ruby Gems and should be available. Any issues, please open a new issue!

**Pull request merged**